code/game/objects/items/implants/implant.dm

```diff
@@ -36,9 +36,7 @@
 
 
 /obj/item/implant/Destroy(force)
-	unimplant()
 	QDEL_NULL(activation_action)
-	part?.implants -= src
```

Comment:
what are you doing

Reply:
So it turns out, unembed_ourselves() will run when the mob gets deleted. So I'm just gonna let that proc handle the unimplanting and removing the implant.